sql/core/src/main/scala/org/apache/spark/sql/execution/commands.scala

```diff
@@ -300,3 +300,72 @@ case class ShowTablesCommand(databaseName: Option[String]) extends RunnableComma
     rows
   }
 }
+
+/**
+ * A command for users to list all of the registered functions.
+ * The syntax of using this command in SQL is:
+ * {{{
+ *    SHOW FUNCTIONS
+ * }}}
+ * :: DeveloperApi ::
+ */
+@DeveloperApi
```

Comment:
can you remove developer api? I think we should just make everything in execution private, and remove developer api from all of them. They were there initially because Michael wanted others to understand analyze, but I don't think that makes a lot of sense anymore.